universal-login-react/src/ui/Onboarding/Onboarding.tsx

```diff
@@ -89,11 +89,11 @@ export const Onboarding = ({emailFlow = false, ...props}: OnboardingProps) => {
         <Route
           path='/code'
           exact
-          render={({history, location}) =>
+          render={({history}) =>
             <ConfirmCodeScreen
               walletService={props.walletService}
               hideModal={props.hideModal}
-              onConfirmCode={() => {history.push(location.state.redirectPath);}}
+              onConfirmCode={() => {history.push(props.walletService.isKind('Confirmed') ? '/create' : '/restore');}}
```

Comment:
I would introduce a new function. e.g. `getRedirectPathForConfirmCode(walletState)`. On other wallet states, we should throw an error.